src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/TableMappingFactory.java

```diff
@@ -101,17 +109,33 @@ void validateCreateVirtualTableRequest(CreateTableRequest createVirtualTableRequ
         virtualTableCreationValidator.validateCompatiblePrimaryKeys(virtualTable, physicalTable);
 
         // validate secondary indexes
-        virtualTableCreationValidator.validateAndGetSecondaryIndexMap(virtualTable, physicalTable);
+        Map<DynamoSecondaryIndex, DynamoSecondaryIndex> secondaryIndexMap =
+            virtualTableCreationValidator.validateAndGetSecondaryIndexMap(virtualTable, physicalTable);
+
+        // create the physical table for a new virtual table, or describe the physical table for an existing virtual
+        // table. set the returned physical table description back onto the table mapping, so it includes things that
+        // can only be determined after the physical table is created, like the streamArn.
+        physicalTable = createPhysicalTableIfNotExists
+            ? physicalTableManager.createTableIfNotExists(physicalCreateTableRequest)
+            : physicalTableManager.describeTable(physicalTable.getTableName());
+
+        TableMapping tableMapping = partitioningStrategy.createTableMapping(virtualTable, physicalTable,
+            secondaryIndexMap::get, mtContext);
+        LOG.debug("created virtual to physical table mapping: " + tableMapping.toString());
+        return tableMapping;
     }
 
     /**
      * Calls the provided CreateTableRequestFactory passing in the virtual table description and returns the
      * corresponding physical table.  Throws a ResourceNotFoundException if the implementation returns null.
      */
-    private DynamoTableDescription lookupPhysicalTable(DynamoTableDescription virtualTable) {
-        return new DynamoTableDescriptionImpl(
-            createTableRequestFactory.getCreateTableRequest(virtualTable).orElseThrow(() ->
-                new ResourceNotFoundException("table " + virtualTable.getTableName() + " is not a supported table")));
+    private CreateTableRequest lookupPhysicalTable(VirtualDynamoTableDescription virtualTable) {
+        // if this is a multitenant table, then create a new physical table dedicated to this purpose.
+        // otherwise, find the corresponding static physical table.
+        return virtualTable.isMultitenant()
```

Comment:
If this is a multitenant table, then create a dedicated physical table for it on the fly. Thought about making this not hard-coded here, and letting CreateTableRequestFactory decide whether to use one of the static tables or create a new physical table, but if we make it more flexible, then deleteTable becomes trickier (if the MT table doesn't necessarily have its own physical table, then we can't just delete the physical table).